src/main/java/kitchenpos/domain/MenuProduct.java

```diff
@@ -1,28 +1,38 @@
 package kitchenpos.domain;
 
+import javax.persistence.*;
+
+@Entity
 public class MenuProduct {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long seq;
-    private Long menuId;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Menu menu;
     private Long productId;
     private long quantity;
 
-    public MenuProduct(Long seq, Long menuId, Long productId, long quantity) {
-        this.seq = seq;
-        this.menuId = menuId;
-        this.productId = productId;
-        this.quantity = quantity;
+    protected MenuProduct() {
     }
 
-    public Long getSeq() {
-        return seq;
+    public MenuProduct(Long productId, long quantity) {
+        this.productId = productId;
+        this.quantity = quantity;
     }
 
-    public Long getMenuId() {
-        return menuId;
+    public void toMenu(Menu menu) {
+        if (menu == null) {
+            throw new IllegalArgumentException();
+        }
+        if (this.menu != null) {
+            this.menu.getMenuProducts().getMenuProducts().remove(this);
+        }
+        this.menu = menu;
+        menu.getMenuProducts().getMenuProducts().add(this);
```

Comment:
```suggestion
        menu.addMenuProduct(this);
```
getter 를 사용하기 보다는 위와 같이 해보면 어떨까요? 😄

Reply:
양방향 연관관계를 맺기 위해 편의상 컬렉션에 직접 접근하여 add를 호출하게 작성하였습니다!

현재 addMenuProduct를 호출하게 되면 메서드끼리 서로 호출하여 무한루프에 빠질 것 같은데..

메서드 순환참조를 하고 조건식으로 무한루프에서 빠져나오는게 더 나을까요..?🤔